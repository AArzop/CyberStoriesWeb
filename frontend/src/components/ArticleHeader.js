import React, { Component } from 'react'
import withStyles from 'react-jss'
import { compose } from 'redux'

const styles = {
  header: {
    fontSize: '29px',
    color: '#555',
    textAlign: 'center',
    marginTop: '2rem',
    marginBottom: '1rem'
  }
}

class ArticleHeader extends Component {
  render () {
    return (
      <h3 className={this.props.classes.header}>{this.props.children}</h3>
    )
  }
}

ArticleHeader.defaultProps = {}

export default compose(
  withStyles(styles)
)(ArticleHeader)
