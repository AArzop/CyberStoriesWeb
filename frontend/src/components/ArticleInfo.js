import React, { Component } from 'react'
import withStyles from 'react-jss'
import { compose } from 'redux'

const styles = {
  info: {
    fontFamily: 'Arial, Helvetica, sans-serif',
    fontWeight: 'bold',
    backgroundColor: '#73d779',
    color: 'white',
    padding: '15px'
  }
}

class ArticleInfo extends Component {
  render () {
    return (
      <div className={this.props.classes.info}>{this.props.children}</div>
    )
  }
}

ArticleInfo.propTypes = {}

ArticleInfo.defaultProps = {}

export default compose(
  withStyles(styles)
)(ArticleInfo)
