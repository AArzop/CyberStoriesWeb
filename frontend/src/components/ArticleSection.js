import React, { Component } from 'react'
import withStyles from 'react-jss'
import { compose } from 'redux'

const styles = {
  section: {
    fontSize: '21px',
    color: '#555',
    textAlign: 'center',
    marginTop: '2rem',
    marginBottom: '1rem'
  }
}

class ArticleSection extends Component {
  render () {
    return (
      <h4 className={this.props.classes.section}>{this.props.children}</h4>
    )
  }
}

ArticleSection.propTypes = {}

ArticleSection.defaultProps = {}

export default compose(
  withStyles(styles)
)(ArticleSection)
