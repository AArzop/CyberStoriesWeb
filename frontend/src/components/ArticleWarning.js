import React, { Component } from 'react'
import withStyles from 'react-jss'
import { compose } from 'redux'

const styles = {
  warning: {
    backgroundColor: '#ee9c41',
    fontFamily: 'Arial, Helvetica, sans-serif',
    fontWeight: 'bold',
    color: 'white',
    padding: '15px'
  }
}

class ArticleWarning extends Component {
  render () {
    return (
      <div className={this.props.classes.warning}>{this.props.children}</div>
    )
  }
}

ArticleWarning.propTypes = {}

ArticleWarning.defaultProps = {}

export default compose(
  withStyles(styles)
)(ArticleWarning)
