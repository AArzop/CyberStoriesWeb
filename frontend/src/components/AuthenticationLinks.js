import React, { Component } from 'react'
import withStyles from 'react-jss'
import { compose } from 'redux'
import { withProfile } from '../actions/authentication'
import LoginRegisterLinks from '../elements/LoginRegisterLinks'
import ProfileDropDown from '../elements/ProfileDropDown'
import Skeleton, { SkeletonTheme } from 'react-loading-skeleton'

const styles = {
  profile: {
    marginLeft: '3px'
  }
}

const skeleton = (
  <span className={'mx-auto'}>
    <SkeletonTheme color={'#111'} highlightColor={'#333'}>
      <Skeleton width={110}/>
    </SkeletonTheme>
  </span>
)

class AuthenticationLinks extends Component {
  render () {
    return (
      this.props.profile ? (
        <ProfileDropDown className={this.props.classes.profile} profile={this.props.profile}/>
      ) : (
        <LoginRegisterLinks/>
      )
    )
  }
}

AuthenticationLinks.propTypes = {}

AuthenticationLinks.defaultProps = {}

export default compose(
  withStyles(styles),
  withProfile(skeleton)
)(AuthenticationLinks)
