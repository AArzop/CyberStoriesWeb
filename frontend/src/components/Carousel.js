import React, { Component } from 'react'
import withStyles from 'react-jss'
import classNames from 'classnames'

const styles = {
  image: {
    width: '100%'
  },
  thumbnail: {
    transition: 'all 0.3s',
    cursor: 'pointer',
    position: 'relative',
    '&:hover': {
      boxShadow: '0 4px 8px 0 rgba(0, 0, 0, 0.2), 0 6px 20px 0 rgba(0, 0, 0, 0.19)'
    }
  }
}

class Carousel extends Component {
  componentDidMount () {
    $(document).ready(function () { //eslint-disable-line
      $('.carousel-' + this.props.uuid) //eslint-disable-line
        .slick({
          slidesToShow: 1,
          slidesToScroll: 1,
          arrows: false,
          fade: true
        })
    }.bind(this))
  }

  render () {
    return (
      <div className={'carousel-' + this.props.uuid}>
        {this.props.images.map((image, index) => (
          <div
            key={index}
            className={classNames(this.props.classes.thumbnail, 'm-3')}
          >
            <img
              src={image}
              className={this.props.classes.image}
            />
          </div>
        ))}
      </div>
    )
  }
}

export default withStyles(styles)(Carousel)
