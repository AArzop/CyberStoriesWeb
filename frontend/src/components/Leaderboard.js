import React, {Component} from 'react';
import '../resources/css/loader.css';
import withStyles from 'react-jss';
import classNames from 'classnames';

const styles = {
    cell: {
        border: '1px solid rgba(100, 100, 100, 0.3)'
    },
    leaderboard: {
        border: '1px solid rgba(100, 100, 100, 0.3)'
    },
    header: {
        fontSize: '17px',
        fontWeight: 'bold'
    }
};

class Leaderboard extends Component {
    constructor(props) {
        super(props);

        this.state = {
            data: {
                loading: false,
                error: null,
                response: null
            }
        };
    }

    componentDidMount() {
        this.setState({
            data: {
                loading: true,
                error: null,
                response: null
            }
        });
        fetch('/leaderboard/list_leaderboard_entries')
            .then(response => response.json())
            .then(json => this.setState({
                data: {
                    loading: false,
                    error: null,
                    response: json
                }
            }))
            .catch(error => this.setState({
                data: {
                    loading: false,
                    error: error,
                    response: null
                }
            }));
    }

    render() {

        if (this.state.data.response) {
            return (
                <div className={classNames('container m-4 w-75 mx-auto', this.props.classes.leaderboard)}>
                    <div className={classNames('row', this.props.classes.header)}>
                        <div className={classNames('col-sm p-1', this.props.classes.cell)}>
                            Pseudo
                        </div>
                        <div className={classNames('col-sm p-1', this.props.classes.cell)}>
                            Score
                        </div>
                    </div>
                    {this.state.data.response.map((entry, index) => (
                        <div className={'row'} key={index}>
                            <div className={classNames('col-sm p-1', this.props.classes.cell)}>
                                {entry.pseudo}
                            </div>
                            <div className={classNames('col-sm p-1', this.props.classes.cell)}>
                                {entry.score}
                            </div>
                        </div>
                    ))}
                </div>
            );
        }

        if (this.state.data.error) {
            return (
                <div>An error has occurred : {String(this.state.error)}</div>
            );
        }

        if (this.state.data.loading) {
            return (
                <div className="lds-ring mt-5">
                    <div/>
                    <div/>
                    <div/>
                    <div/>
                </div>
            );
        }

        return null;
    }
}

export default withStyles(styles)(Leaderboard);