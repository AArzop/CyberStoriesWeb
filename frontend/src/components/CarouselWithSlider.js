import React, { Component } from 'react'
import withStyles from 'react-jss'
import Carousel from './Carousel'
import Slider from './Slider'
import computers from '../resources/images/computers.jpg'
import computers2 from '../resources/images/computers2.jpeg'

const styles = {}

class CarouselWithSlider extends Component {
  constructor (props) {
    super(props)
    this.state = {
      images: this.props.images,
      current: 0
    }

    this.handleImageChange = this.handleImageChange.bind(this)
  }

  handleImageChange (index) {
    this.setState({ current: index })
  }

  render () {
    return (
      <div style={{ width: 700 }} className={this.props.className}>
        <Carousel uuid={this.props.uuid} images={this.state.images}/>
        <Slider uuid={this.props.uuid} images={this.state.images}
          beforeChange={this.handleImageChange}/>
      </div>
    )
  }
}

CarouselWithSlider.defaultProps = {
  images: [computers, computers2], // list of urls
  uuid: 0
}

export default withStyles(styles)(CarouselWithSlider)
