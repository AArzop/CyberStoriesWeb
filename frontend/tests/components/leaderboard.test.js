import React from 'react'
import { Leaderboard } from '../../src/components/Leaderboard'
import renderer from 'react-test-renderer'
import { MemoryRouter } from 'react-router'

test('if leaderboard and prints data from prop', () => {
  const leaderboardEntriesMock = [
    {
      id: 0,
      pseudo: 'John',
      email: 'john.doe@gmail.com',
      gameResults: [
        {
          id: 0,
          date: '2019/05/01',
          stage1Score: 10,
          stage2Score: 20,
          stage3Score: 30,
          stage4Score: 40,
          visible: true,
          player: 0
        }
      ]
    },
    {
      id: 1,
      pseudo: 'Ann',
      email: 'ann.smith@gmail.com',
      gameResults: [
        {
          id: 1,
          date: '2019/05/02',
          stage1Score: 20,
          stage2Score: 30,
          stage3Score: 40,
          stage4Score: 50,
          visible: true,
          player: 1
        }
      ]
    }
  ]

  const component = renderer.create(
    <MemoryRouter>
      <Leaderboard
        leaderboardEntries={leaderboardEntriesMock}
      />
    </MemoryRouter>
  )
  const json = component.toJSON()
  expect(json).toMatchSnapshot()
})
