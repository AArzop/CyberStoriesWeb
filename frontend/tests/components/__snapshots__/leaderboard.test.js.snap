// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`if leaderboard and prints data from prop 1`] = `
<div
  className="m-4 w-75 mx-auto"
>
  <br />
  <div
    className="m-3 PitchTitle-pitchTitle-0-1-1 text-center"
  >
    Leaderboard
  </div>
  <br />
  <div
    className="container"
  >
    <div
      className="row"
    >
      <div
        className="col-sm p-1"
      >
        Pseudo
      </div>
      <div
        className="col-sm p-1"
      >
        Best average score
      </div>
      <div
        className="col-sm p-1"
      >
        Number of games
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-sm p-1"
      >
        <a
          aria-current={null}
          href="/leaderboard/?player_id=0"
          onClick={[Function]}
        >
          John
        </a>
      </div>
      <div
        className="col-sm p-1"
      >
        <div
          className="GrowingBar-wrapper-0-1-3"
        >
          <div
            className="m-1 GrowingBar-growingBar-0-1-2"
            style={
              Object {
                "backgroundColor": "#ff2a2a",
                "width": 0,
              }
            }
          />
          <div
            style={
              Object {
                "color": "#ff2a2a",
                "fontWeight": "bold",
              }
            }
          >
            0
             %
          </div>
        </div>
      </div>
      <div
        className="col-sm p-1"
      >
        1
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col-sm p-1"
      >
        <a
          aria-current={null}
          href="/leaderboard/?player_id=1"
          onClick={[Function]}
        >
          Ann
        </a>
      </div>
      <div
        className="col-sm p-1"
      >
        <div
          className="GrowingBar-wrapper-0-1-3"
        >
          <div
            className="m-1 GrowingBar-growingBar-0-1-2"
            style={
              Object {
                "backgroundColor": "#ff2a2a",
                "width": 0,
              }
            }
          />
          <div
            style={
              Object {
                "color": "#ff2a2a",
                "fontWeight": "bold",
              }
            }
          >
            0
             %
          </div>
        </div>
      </div>
      <div
        className="col-sm p-1"
      >
        1
      </div>
    </div>
  </div>
  <br />
  <br />
  <br />
  <input
    onChange={[Function]}
    placeholder="Search pseudo"
    type="text"
    value=""
  />
  <p>
    Click on a pseudo to see score details.
  </p>
</div>
`;
